internal/handler/users_handler.go

```diff
@@ -79,3 +80,99 @@ func GetUsersHandler(store *db.Store) gin.HandlerFunc {
 		ctx.JSON(http.StatusOK, user)
 	}
 }
+
+// ListUsers handler
+type listUsersRequest struct {
+	Limit  int32 `form:"limit" binding:"required,min=1"`
+	Offset int32 `form:"offset" binding:"required,min=0"`
+}
+
+func ListUsersHandler(store *db.Store) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var req listUsersRequest
+		if err := ctx.ShouldBindQuery(&req); err != nil {
+			ctx.JSON(http.StatusBadRequest, errorResponse(err))
+			return
+		}
+
+		args := db.ListUsersParams{
+			Limit:  req.Limit,
+			Offset: req.Offset,
+		}
```

Comment:
Because we using the `page_size` and `page_id` you'll need to update 
the params you are passing to:

```go
Limit:  req.PageSize,
Offset: (req.PageID - 1) * req.PageSize,
 ```

Reply:
Ready to check again.